server/lobbyconnection.py

```diff
@@ -653,18 +657,32 @@ def decodeUniqueId(self, serialized_uniqueid):
 
         return True
 
+    def clean_version(self):
+        if self.version is not None:
+            return self.version.split('-')[0].split('+')[0]
+        else:
+            return None
+
+    def statsd_gauge(self, delta):
+        vgauge = self.STATSD_VERSION.format(self.user_agent, self.version.replace('.','_'))
+        self._logger.debug('updating version gauge string: {} ({})'.format(vgauge, delta))
+        server.stats.gauge(vgauge, delta, delta=True)
+        server.stats.gauge(self.STATSD_USERAGENT.format(self.user_agent), delta, delta=True)
+
     def check_version(self, message):
         versionDB, updateFile = self.player_service.client_version_info
         update_msg = dict(command="update",
                           update=updateFile,
                           new_version=versionDB)
 
+
         self.user_agent = message.get('user_agent')
-        version = message.get('version')
-        server.stats.gauge('user.agents.None', -1, delta=True)
-        server.stats.gauge('user.agents.{}'.format(self.user_agent), 1, delta=True)
+        self.version = message.get('version')
+        clean_version = self.clean_version()
+        # revert the "None" user agent set in servercontext
+        self.statsd_gauge(1)
```

Comment:
Since we set `self.user_agent` above, this won't actually clean the `None` gauge? I presume the comment is wrong, taking the later changes into account.